x-pack/plugins/actions/server/types.ts

```diff
@@ -101,11 +102,12 @@ export type ExecutorType<
   options: ActionTypeExecutorOptions<Config, Secrets, Params>
 ) => Promise<ActionTypeExecutorResult<ResultData>>;
 
-export interface ValidatorType<Type> {
+export interface ValidatorType<T> {
   schema: {
-    validate(value: unknown): Type;
+    validate(value: unknown): T;
+    getSchema?: () => AnySchema;
```

Comment:
Why is this typed as optional?

Reply:
I changed it then remembered why i did it so.
1- SubActionsConnectorType converts its definition to ConnectorType def and misses that field (We can fix this I think)
2- There are hundreds of tests cases with ActionType mock without this filed :) And we are using it only in this test, so i made it optional.